src/gateio_new_coins_announcements_bot/announcement_scrapers/kucoin_scraper.py

```diff
@@ -0,0 +1,46 @@
+import random
+import time
+import requests
+
+from gateio_new_coins_announcements_bot.logger import logger
+from gateio_new_coins_announcements_bot.util.random import random_str, random_int
+
+
+class KucoinScraper:
+    def __init__(self, http_client=requests):
+        self.http_client = http_client
+
+    def fetch_latest_announcement(self):
```

Comment:
This is really similar to `BinanceScraper.fetch_latest_annoucement`.
Maybe we can somehow generalize this and pass in the query as a parameter?
And use inheritance with a `Scraper` base class and the individual Binance/Kucoin-Scrapers just implement the _request_url method and handle the json response via a _extract_announcement method?
I'm not entirely sure on this, so other ideas are welcomed :)

Reply:
I like your idea of adding a `_extract_announcement` method. However, I'm trying not to overengineer that part just yet, and want to see how the codebase evolves. A little duplication is OK for the moment, but when we add more scrapers we can tackle this

Reply:
We will keep it in mind for later then, fine by me for now